lib/Zonemaster/Backend/RPCAPI.pm

```diff
@@ -41,7 +41,10 @@ sub new {
             die "$@ \n" if $@;
             $self->{db} = "$params->{db}"->new();
         };
-        die "$@ \n" if $@;
+        if ($@) {
+            warn "Failed to initialize the [$params->{db}] database backend module: [$@] \n";
+            die "Failed to initialize the [$params->{db}] database backend module \n";
```

Comment:
What will go back to the client? What will go a log file?

Reply:
The warn goes to the log file.
The die goes to the API client.

Reply:
I feel that the message to the API client is too long and contains too much information. I suggest "INTERNAL SYSTEM ERROR" plus some code, and the same code in the log message. That code could be a fixed code "INTERNAL SYSTEM ERROR # 5011" or more like a ticket ID which could be a random number that goes into both messages.

Reply:
If this becomes a JSON-RPC error response with code -32603 (which seems appropriate) the message should be `Internal error` as per the [JSON-RPC specification](https://www.jsonrpc.org/specification#error_object).

This goes for the other calls to `die` in this sub and in `handle_exception` as well.

Reply:
Fixed in commit 416b48b